File: skyspark/src/main/java/com/github/dkanellis/skyspark/api/helpers/TextFileToPointRDD.java
package com.github.dkanellis.skyspark.api.helpers;

import com.github.dkanellis.skyspark.api.utils.point.Points;
import org.apache.spark.api.java.JavaRDD;

import java.awt.geom.Point2D;

public class TextFileToPointRDD {

    private final SparkContextWrapper sparkContext;

    public TextFileToPointRDD(SparkContextWrapper sparkContext) {
        this.sparkContext = sparkContext;
    }

    public JavaRDD<Point2D> getPointRDDFromTextFile(String filePath, String delimiter) {
        JavaRDD<String> lines = sparkContext.textFile(filePath);
        JavaRDD<Point2D> points = convertToPoints(lines, delimiter);
        return points;
    }

    private JavaRDD<Point2D> convertToPoints(JavaRDD<String> lines, String delimiter) {
        return lines.map(line -> Points.pointFromTextLine(line, delimiter));
    }
}

File: skyspark-scala/src/main/scala/com/github/dkanellis/skyspark/scala/api/helpers/TextFileToPointRdd.scala
package com.github.dkanellis.skyspark.scala.api.helpers

import com.github.dkanellis.skyspark.scala.api.algorithms.Point
import org.apache.spark.SparkContext

object TextFileToPointRdd {

  def convert(sparkContext: SparkContext, filePath: String, delimiter: String) = {
    sparkContext.textFile(filePath)
      .map(pointFromTextLine(_, delimiter))
  }

  private def pointFromTextLine(textLine: String, delimiter: String) = {
    val numberArray = textLine
      .trim
      .split(delimiter)
      .map(_.toDouble)
    new Point(numberArray: _*)
  }
}
